src/template.rs

```diff
@@ -0,0 +1,137 @@
+use crate::Event;
+use bytes::Bytes;
+use lazy_static::lazy_static;
+use regex::bytes::{Captures, Regex};
+use string_cache::DefaultAtom as Atom;
+
+lazy_static! {
+    static ref RE: Regex = Regex::new(r"\{\{(?P<key>[^\}]+)\}\}").unwrap();
+}
+
+#[derive(Debug, Clone)]
+pub struct Template {
+    src: Bytes,
```

Comment:
I think it would be nice to precompute if there is even a regex match and have a bool flag here to bypass the replace_all if there is not one match.